plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,633 @@
+        }
+      ]
+    }
+  }
+'''
+
+import json
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.compat.version import LooseVersion
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import (
+    get_idrac_firmware_version, get_dynamic_uri, validate_and_get_first_resource_id_uri, remove_key, idrac_redfish_job_tracking)
+from datetime import datetime
+
+MANAGERS_URI = "/redfish/v1/Managers"
+IDRAC_JOB_URI = "{res_uri}/Jobs/{job_id}"
+
+OEM = "Oem"
+MANUFACTURER = "Dell"
+LC_SERVICE = "DellLCService"
+ACTIONS = "Actions"
+EXPORT = "#DellLCService.ExportePSADiagnosticsResult"
+RUN = "#DellLCService.RunePSADiagnostics"
+
+SUCCESS_EXPORT_MSG = "Successfully exported the diagnostics."
+EXPORT_NO_RESULTS_MSG = "Unable to export the diagnostics results because the results do not exist."
+SUCCESS_RUN_MSG = "Successfully run the diagnostics operation."
+SUCCESS_RUN_AND_EXPORT_MSG = "Successfully run and exported the diagnostics."
+RUNNING_RUN_MSG = "Successfully triggered the job to run diagnostics."
+INVALID_FILE_MSG = "File extension is invalid. Supported extension is: .txt."
+INVALID_DIRECTORY_MSG = "Provided directory path '{path}' is not valid."
+INSUFFICIENT_DIRECTORY_PERMISSION_MSG = "Provided directory path '{path}' is not writable. " \
+                                        "Please check if the directory has appropriate permissions"
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The parameter `job_wait_timeout` value cannot be negative or zero."
+WAIT_TIMEOUT_MSG = "The job is not complete after {0} seconds."
+START_TIME = "The specified scheduled start time occurs in the past, " \
+             "provide a future time to schedule the start time."
+INVALID_TIME = "The specified date and time `{0}` to schedule the diagnostics is not valid. Enter a valid date and time."
+END_START_TIME = "The end time `{0}` to schedule the diagnostics must be greater than the start time `{1}`."
+
+IGNORE_CERTIFICATE_WARNING = {"off": "Off", "on": "On"}
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class ExportDiagnostics:
+    STATUS_SUCCESS = [200, 202]
+
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        try:
+            share_type = module.params.get('share_parameters').get('share_type')
+            export_diagnostics_url = self.__get_export_diagnostics_url(module)
+            job_status = {}
+            self.__check_file_extension(module)
+            if share_type == "local":
```

Comment:
1. check if you can remove if else ladder, use dictionary or local()/globals()
2. job_status is common in each if statement call that outside of ladder.

Reply:
done